tests/ui/drop/lint-if-let-rescope-gated.rs

```diff
@@ -0,0 +1,32 @@
+// This test checks that the lint `if_let_rescope` does not action if either
```

Comment:
@jieyouxu I also added the test to show that the lint is behind the `if_let_rescope` gate and the Edition 2024 gate.